config/routes.js

```diff
@@ -1,56 +1,49 @@
- *
- * Note: Sails doesn't ACTUALLY serve stuff from `assets`-- the default Gruntfile in Sails copies
- * flat files from `assets` to `.tmp/public`.  This allows you to do things like compile LESS or
- * CoffeeScript for the front-end.
- *
- * For more information on configuring custom routes, check out:
- * http://sailsjs.org/#/documentation/concepts/Routes/RouteTargetSyntax.html
- */
-
 module.exports.routes = {
-
-  /***************************************************************************
-  *                                                                          *
-  * Custom routes here...                                                    *
-  *                                                                          *
-  *  If a request to a URL doesn't match any of the custom routes above, it  *
-  * is matched against Sails route blueprints. See `config/blueprints.js`    *
-  * for configuration options and examples.                                  *
-  *                                                                          *
-  ***************************************************************************/
-
-  'post /webhook/github': 'WebhookController.github',
-
-  'post /v0/build/:id/status/:token': 'BuildController.status',
-  'post /v0/build/:id/restart': 'BuildController.restart',
-
-  'get /preview/:owner/:repo/:branch': 'PreviewController.proxy',
-  'get /preview/:owner/:repo/:branch/*': 'PreviewController.proxy',
-
-  'get /auth/github': 'AuthController.github',
-  'get /auth/github/callback': 'AuthController.callback',
-  'get /logout': 'AuthController.logout',
-
-  'get /v0/me': 'UserController.me',
-
-  'post /v0/build/:build_id/log/:token': 'BuildLogController.create',
-  'get  /v0/build/:build_id/log': 'BuildLogController.find',
-
-  'get /sites(/*)?': {
-    controller: 'main',
-    action: 'index'
-  }
-};
+  /**
+    API
+  */
+  // Builds
+  "get /v0/build": "BuildController.find",
+  "get /v0/build/:id": "BuildController.findOne",
+  "post /v0/build/:id/restart": "BuildController.restart",
+
+  // Build logs
+  "get /v0/build/:build_id/log": "BuildLogController.find",
+
+  // Sites
+  "get /v0/site": "SiteController.find",
+  "post /v0/site": "SiteController.create",
+  "get /v0/site/:id": "SiteController.findOne",
+  "put /v0/site/:id": "SiteController.update",
+  "delete /v0/site/:id": "SiteController.destroy",
+
+  // Users
+  "get /v0/me": "UserController.me",
+  "get /v0/user/usernames": "UserController.usernames",
+
+  /**
+    User exposed routes
+  */
+
+  // Auth
+  "get /auth/github": "AuthController.github",
+  "get /auth/github/callback": "AuthController.callback",
+  "get /logout": "AuthController.logout",
+
+  // Frontend
+  "get /sites(/*)?": "MainController.index",
+
+  // Previews
+  "get /preview/:owner/:repo/:branch(/*)?": "PreviewController.proxy",
+
+  /**
+    Webhooks
+  */
+  "post /webhook/github": "WebhookController.github",
```

Comment:
Would it make sense to version the webhook endpoint? Eh... It's internal as far as the "clients" are concerned.